lib/src/signed_video_h26x_sign.c

```diff
@@ -224,21 +221,32 @@ generate_sei_nalu(signed_video_t *self, uint8_t **payload, uint8_t **payload_sig
   SVI_TRY()
     // Get the total payload size of all TLVs. Then compute the total size of the SEI NALU to be
     // generated. Add extra space for potential emulation prevention bytes.
-    document_size = tlv_list_encode_or_get_size(self, document_encoders, num_doc_encoders, NULL);
+    optional_document_size = tlv_list_encode_or_get_size(
+        self, optional_document_encoders, num_optional_document_encoders, NULL);
+    mandatory_document_size = !self->is_golden_sei
+        ? tlv_list_encode_or_get_size(self, mandatory_document_encoders, num_man_doc_encoders, NULL)
+        : 0;
     gop_info_size = tlv_list_encode_or_get_size(self, gop_info_encoders, num_gop_encoders, NULL);
     if (self->num_vendor_encoders > 0 && self->vendor_encoders) {
       vendor_size =
           tlv_list_encode_or_get_size(self, self->vendor_encoders, self->num_vendor_encoders, NULL);
     }
 
+    document_size = optional_document_size + mandatory_document_size;
     payload_size = document_size + gop_info_size + vendor_size;
     payload_size += UUID_LEN;  // UUID
     payload_size += 1;  // One byte for reserved data.
     if ((self->max_sei_payload_size > 0) && (payload_size > self->max_sei_payload_size)) {
```

Comment:
Add a `&& mandatory_document_size > 0` in the if-statement, since it only affects the mandatory part. This way you do not need tho check the `is_golden_sei` flag inside either.

Reply:
If statement gets too long if I do that

Reply:
But it is better than entering here and then calculate the size again and after that set it to zero (again). Simply waste of operations.